src/py/_pyodide/_importhook.py

```diff
@@ -131,3 +131,41 @@ def register_js_finder() -> None:
     global JsProxy
     JsProxy = _pyodide_core.JsProxy
     sys.meta_path.append(jsfinder)
+
+
+class StdlibFinder(MetaPathFinder):
+    def __init__(self) -> None:
+        # `test`` is not a stdlib module, but we unvendors in anyway.
+        self.stdlibs = sys.stdlib_module_names | {"test"}
+
+        # TODO: put list of unvendored stdlibs to somewhere else?
+        self.unvendored_stdlibs = {"distutils", "test", "_ssl", "_lzma"} & self.stdlibs
+
+    def find_spec(
+        self,
+        fullname: str,
+        path: Sequence[bytes | str] | None,
+        target: ModuleType | None = None,
+    ) -> ModuleSpec | None:
+        [parent, _, _] = fullname.partition(".")
+
+        if not parent or parent in sys.modules or parent not in self.stdlibs:
+            return None
+
+        if parent in self.unvendored_stdlibs:
+            raise ModuleNotFoundError(
+                f"The module '{parent}' is unvendored from Pyodide stdlib, "
+                f'you can install it by calling: await pyodide.loadPackage("{parent}"). '
```

Comment:
Let's maybe rather use micropip (and assume it's either loaded or will be loaded via import), because loadPackage needs to be called from JS while most users will use this from Python and can only write Python.

Reply:
You could also say `pyodide_js.loadPackage` here. `from pyodide_js import loadPackage;  await loadPackage("some_package")` will work even if `micropip` isn't around.

Reply:
Changed to `micropip.install`. Maybe we can mention other ways in docs.